mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -66,7 +73,12 @@ public HiveAuthorizationProvider getAuthorizationProvider() throws HiveException
 
   @Override
   public void configureInputJobProperties(TableDesc tableDesc, Map<String, String> map) {
+    Properties props = tableDesc.getProperties();
+    Table table = Catalogs.loadTable(conf, props);
 
+    map.put(InputFormatConfig.TABLE_IDENTIFIER, props.getProperty(NAME));
+    map.put(InputFormatConfig.TABLE_LOCATION, table.location());
+    map.put(InputFormatConfig.TABLE_SCHEMA, SchemaParser.toJson(table.schema()));
```

Comment:
Doesn't this also need to forward the catalog loader class?

Reply:
Depends upon where we set the catalog loader class.   If we set it in HiveConf, then it will always be available by default in configuration. If we want to expose catalog loader class as a table property, e.g to load different tables differently, then we can forward it here.

Reply:
It seems like it would always comes from table properties to me, since we might have tables from different Iceberg catalogs.

Reply:
Yes, and ideally we'd have a unit test which uses multiple tables which are set up differently and checks that the properties that are unique to each table are honoured properly.

Reply:
The current implement of the catalog loader reads directly from the HiveConf and ignores the table properties, that why the catalog loader class is not forwarded here.

I'm planning to add the ability to declare the catalog in the table properties but I'd rather do so in a follow-up PR because that requires a refactor of the current test suite. I'll make sure to add a test that reads from multiples tables set up with different catalogs.